src/s3/commands/uploadFile.ts

```diff
@@ -114,11 +148,221 @@ async function uploadWithProgress({
     )
 }
 
-function statFile(file: vscode.Uri) {
-    return statSync(file.fsPath).size
+interface BucketQuickPickItem extends vscode.QuickPickItem {
+    bucket: S3.Bucket | undefined
 }
 
-async function refreshNode(node: S3BucketNode | S3FolderNode, commands: Commands): Promise<void> {
-    node.clearChildren()
-    return commands.execute('aws.refreshAwsExplorerNode', node)
+/**
+ * 
+ * @returns 
+ */
+async function promptUserForBucket(): Promise<S3.Bucket | undefined> {
+    let regionCode = Workspace.vscode().getConfiguration(extensionSettingsPrefix).get<string>('s3.defaultRegion')?? "us-east-1" //TODO:: decide how to get the regionCode!!
+    const s3client = ext.toolkitClientBuilder.createS3Client(regionCode)
+    const window = Window.vscode()
+    const allBuckets: S3.Bucket[] = await s3client.listAllBuckets()
```

Comment:
What happens if this call fails? You should `try/catch` from here to the end of the function, and that should result in an upload or list buckets failure for telemetry purposes.

Reply:
Just surrounding the call to promptUserForBucket() in a try catch should do it. Since listAllBuckets() already has a try/catch inside, maybe add telemetry there?